### ISSUE TITLE
`ui_event` when accessibility elements change

### ISSUE THREAD

**Issue by phillco** [talonvoice/talon#491]
This came up in the discussion today about applications not immediately reflecting mutations in `focused_element().attrs`, and how to work around that. But this could also be a handy way to power additional functionality, like updating lists when the application tree changes.

(For example, if we have a list of Slack channels based on the accessibility elements, we could refresh it when that list actually changes, rather than just relying on element_focus / window_title / app_focus / window_focus events, although those get us pretty close)

**Comment by phillco** [comment]
I saw an old Dragon demo video where they supported "go back" after making a correction to return the cursor to its previous position. I was hoping macOS had a keyboard shortcut to navigate the cursor history in a text area, but it doesn't look like they do.

I'm sure their version only works if you make the correction using Dragon, but it struck me that if we had something like `kAXSelectedTextChangedNotification`s we could build cursor position history for every native text field in Talon, and allow you to navigate backwards and forwards.

We can obviously do a very basic version that only tracks voice insertions, but having a native callback that would work regardless of how the cursor position was changed (voice, mouse, keyboard, etc) would be pretty sweet.

**Comment by phillco** [comment]
Chatted with @lunixbochs -- we did have a prototype build last year that did this for macOS, but there were so many notifications, performance was a problem, and had to be reverted.
